### PR TITLE
Rename kusk references to kusk-gen

### LINKED ISSUE #206: The migration to Kuskgen
**Describe the enhancement you'd like to see**

To avoid any confusion about the relationship between Kusk and Kusk Gateway, let's rename Kusk to Kuskgen.
This involves:
1. Renaming the repository path.
2. Fixing the documentation and the official site (including kusk.kubeshop.io).
3. Renaming the binary and package name.
4. Fixing all imports, code comments, symbol names.

### PR DESCRIPTION
**Pull request by kylehodgetts**
This PR closes #206 

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [x] updated the docs
- [ ] added a test

### CODE REVIEW

File: docs/argocd.md [hunk 1]
```diff
@@ -33,7 +33,7 @@ We will add an `initContainer` to `argocd-repo-server` and a volume to download
 +          image: alpine:3.8
 +          command: [ sh, -c ]
 +          args:
-+            - wget -qO- https://github.com/kubeshop/kusk/releases/download/0.0.1-rc1/kusk_0.0.1-rc1_Linux_x86_64.tar.gz | tar -xvzf - &&
++            - wget -qO- https://github.com/kubeshop/kusk-gen/releases/download/0.0.1-rc1/kusk_0.0.1-rc1_Linux_x86_64.tar.gz | tar -xvzf - &&
```
Comment (Tarick) [inline docs/argocd.md:36]: kusk_0.0.1 - probably need some way to specify latest and it will be kusk-gen

**Review by Tarick — CHANGES_REQUESTED** [review]

File: docs/argocd.md (hunk 1 above)
Comment (kylehodgetts) [reply docs/argocd.md:36]: any ideas?

Comment (dobegor) [reply docs/argocd.md:36]: @kylehodgetts https://github.com/kubeshop/kusk/releases/latest <- this always points to the latest release, same mechanic works for download URLs

**Review by Tarick — APPROVED** [review]

**Review by dobegor — APPROVED** [review]

**Comment by jasmingacic** [comment]
Should we merge this?

**Review by jasmingacic — APPROVED** [review]

**Pull request merged**